Presentations/02-intro-to-R-programming.Rmd

```diff
@@ -771,99 +785,48 @@ knitr::include_app("https://rrmaximiliano.shinyapps.io/learnr-app/")
 
 ---
 
-# Looping
-
-## Loops vs Apply
+# Map vs looping
 
 * When looping, you repeat the same operation over a set of items
 
-* `apply()`, instead, takes all your elements at once and applies an operation to them simultaneously
+* `map()`, instead, takes all your elements at once and applies an operation to them simultaneously
 
 * The difference is like this:
 
   + Imagine you ask a yes/no question to a group of people
   + You can collect the answers by asking each one of them individually -- __this is looping__
-  + Otherwise, you can ask them to raise their hands and collect all answers at once -- __this is `apply()`__
+  + Otherwise, you can ask them to raise their hands and collect all answers at once -- __this is `map()`__
 
 * The output of a loop is the regular output of the operation you're repeating, times the number of iterations you did
 
-* The output of `apply()` will be a vector most of times, but it can also be a list or an array of elements (a matrix)
-
----
-
-# Looping
-
-## Apply
-
-* A more general version of `sapply()` is the `apply()` function. This is its syntax:
-
-```{r, eval=FALSE}
-apply(X, MARGIN, FUN, ...)
-```
-
-* Arguments:
-
-  + **X:** a data frame (or matrix) the function will be applied to
-  + **MARGIN:** 1 to apply the function to all rows or 2 to apply the function to all columns
-  + **FUN:** the function you want to apply
-
-* `apply()` applies a function (`FUN`) to all columns or rows of matrix (`X`). A value of 1 in `MARGIN` indicates that the funcion should be applied row-wise, while 2 indicates columns
+* The output of `map()` will be always a list
 
 ---
 
-# Looping
+# Map
 
+.exercise[
 
-```{r, eval = FALSE}
-matrix <- matrix(c(1, 24, 9, 6, 9, 4, 2, 74, 2), nrow = 3) # Defining a matrix
-apply(matrix, 1, mean) # row means
-apply(matrix, 2, mean) # column means
-```
-
-```{r echo=FALSE}
-knitr::include_app("https://rrmaximiliano.shinyapps.io/learnr-app/")
-```
-
----
-
-# Looping
-
-### Exercise 6: Get the row max
+### Exercise 6 `r fa("keyboard")`
 
 1. Select the columns `freedom` and `happiness_score` of `whr`
+2. Use `map()` to get the maximum value of these columns
 
-2. Use `apply()` to get the row max between these two columns, for every row
+> Hints:
+  + Remember the syntax of `map()`: `map(X, function)`. You can also try using pipes here
+  + The function to get the maximum over a set of numbers is `max`.
 
-* Hints:
-  + Remember the syntax of `apply()`: `apply(X, MARGIN, FUN)`
-  + A value of 1 for `MARGIN` indicates that the function must applied row-wise
-  + The function to get the maximum over a set of number is `max`
+]
```

Comment:
There's a slide missing with the solution to this exercise

Reply:
this is for exercise 6, right? thanks for catching this. I added the solution in 185dcc8